File: dev-docs/snippets/blob-buckets/get-signed-url.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Mint a time-limited signed URL that downloads a blob without auth.
// `expiresInSeconds` is optional and defaults server-side.
export async function getSignedUrl(
  client: JsBaoClient,
  bucketIdOrKey: string,
  blobId: string,
) {
  // #region example
  const signed = await client.blobBuckets.getSignedUrl(
    bucketIdOrKey,
    blobId,
    3600,
  );
  const url = signed.url;
  // #endregion example
  return url;
}

File: dev-docs/snippets/blob-buckets/get-signed-url.swift
import JsBaoClient

// Mint a time-limited signed URL that downloads a blob without auth.
// Swift returns a typed `BlobSignedUrlResult` — read `signed.url`,
// `signed.token`, `signed.expiresAt`, `signed.expiresInSeconds`.
func getSignedUrl(
  client: JsBaoClient,
  bucketIdOrKey: String,
  blobId: String
) async throws {
  // #region example
  let signed = try await client.blobBuckets.getSignedUrl(
    bucketIdOrKey: bucketIdOrKey,
    blobId: blobId,
    expiresInSeconds: 3600
  )
  let url = signed.url
  // #endregion example
  _ = url
}
